app/main.py

```diff
@@ -1,8 +1,18 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = {p_data["name"]: Person(p_data["name"], p_data["age"]) for p_data in people}
+    for p_data in people:
+        person = persons[p_data["name"]]
+        if "wife" in p_data and p_data["wife"]:
+            person.wife = persons[p_data["wife"]]
```

Comment:
When assigning `person.wife`, ensure that the `p_data['wife']` exists in the `persons` dictionary to avoid a KeyError. Consider adding a check to ensure the referenced person exists.